samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -26,12 +26,10 @@
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-
-
   synchronized public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
     ZkClient zkClient = new ZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
-    ZkUtils zkUtils = new ZkUtils(participantId, new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
```

Comment:
Good question. When I created ZkUtils, I wanted it to be a util class for interacting with Zookeeper. Hence, I wanted it to be stateless. 

It has somehow evolved into a stateful class (with processorId, which is what is being named as participantId). I have urged @sborya to put up a SEP for discussing the design of the CoordinationService because I don't fully agree with how it has been implemented. Due to lack of documentation, I am unable to answer all your questions with clarity.

So to answer your questions:

> What's this participantId?

Not sure why Boris called it that. But it seems to be the same as processorID.
> Are we changing the znode path after removing the participantId from the parameter? 
No. This "participantId" was only used in log statements.  It doesn't affect the coordination implementation in anyway.

All I wanted to make sure was that processorId is not being treated as a state in ZkUtils. Iirc, processorId was being used in the log statements only. Hence, I felt it is ok to remove it.